UnityProject/Assets/Scripts/Player/PlayerSync.Server.cs

```diff
@@ -596,4 +600,30 @@ private void CheckMovementServer()
 			TryUpdateServerTarget();
 		}
 	}
+
+	private void OnEnable()
+	{
+		onTileReached.AddListener(Cross);
+	}
+	private void OnDisable()
+	{
+		onTileReached.RemoveListener(Cross);
+	}
+
+	private void Cross(Vector3Int position)
+	{
+		registerPlayer.CheckTileSlip();
+
+		if (PlayerUtils.IsGhost(gameObject))
+		{
+			return;
+		}
+		List<GameObject> objects = UITileList.GetItemsAtPosition(position);
```

Comment:
don't use UITileList, it's crap
just use MatrixManager.GetAt<RegisterItem>(pos) and you'll get a nice list of RegisterItems, no need to call GetComponent<> yourself

Reply:
Could we deprecate UITileList so people stop using it? Perhaps we could add a warning to it

Reply:
yeah, totally